src/main/java/aitsi/m3spin/spafrontend/parser/Parser.java

```diff
@@ -219,10 +221,17 @@ private void parseKeyword(EntityType keyword) throws MissingCodeEntityException
 
     private StatementList parseStmtList() throws SimpleParserException {
         List<Statement> stmtList = new ArrayList<>();
-        stmtList.add(parseStmt());
+        this.counter+=1;
+
+        Statement stmt = parseStmt();
+        stmt.setStmtLine(this.counter);
```

Comment:
Zamiast pisać jak wyżej `this.counter+= 1` można jako argument wpisać `++counter`, a samo ustawianie numeru linii powinno zostać przeniesione do metody `parseStmt`

Reply:
Ja zrobię to drugie w wolnej chwili, jeszcze nie scalajcie